libavfilter/dnn/dnn_backend_tf.c

```diff
@@ -906,71 +902,98 @@ static DNNReturnType execute_model_tf(TFRequestItem *request, Queue *inference_q
+        case DFT_PROCESS_FRAME:
+            //it only support 1 output if it's frame in & frame out
+            if (task->do_ioproc) {
+                if (tf_model->model->frame_post_proc != NULL) {
+                    tf_model->model->frame_post_proc(task->out_frame, outputs, tf_model->model->filter_ctx);
+                } else {
+                    ff_proc_from_dnn_to_frame(task->out_frame, outputs, ctx);
+                }
             } else {
-                ff_proc_from_dnn_to_frame(task->out_frame, outputs, ctx);
+                task->out_frame->width = outputs[0].width;
+                task->out_frame->height = outputs[0].height;
             }
-        } else {
-            task->out_frame->width = outputs[0].width;
-            task->out_frame->height = outputs[0].height;
-        }
-        break;
-    case DFT_ANALYTICS_DETECT:
-        if (!tf_model->model->detect_post_proc) {
-            av_log(ctx, AV_LOG_ERROR, "Detect filter needs provide post proc\n");
+            break;
+        case DFT_ANALYTICS_DETECT:
+            if (!tf_model->model->detect_post_proc) {
+                av_log(ctx, AV_LOG_ERROR, "Detect filter needs provide post proc\n");
+                return DNN_ERROR;
+            }
+            tf_model->model->detect_post_proc(task->out_frame, outputs, task->nb_output, tf_model->model->filter_ctx);
+            break;
+        default:
+            for (uint32_t i = 0; i < task->nb_output; ++i) {
+                if (infer_request->output_tensors[i]) {
+                    TF_DeleteTensor(infer_request->output_tensors[i]);
+                }
+            }
+            av_log(ctx, AV_LOG_ERROR, "Tensorflow backend does not support this kind of dnn filter now\n");
             return DNN_ERROR;
         }
-        tf_model->model->detect_post_proc(task->out_frame, outputs, task->nb_output, tf_model->model->filter_ctx);
-        break;
-    default:
         for (uint32_t i = 0; i < task->nb_output; ++i) {
             if (infer_request->output_tensors[i]) {
                 TF_DeleteTensor(infer_request->output_tensors[i]);
             }
         }
-        av_log(ctx, AV_LOG_ERROR, "Tensorflow backend does not support this kind of dnn filter now\n");
-        return DNN_ERROR;
-    }
-    for (uint32_t i = 0; i < task->nb_output; ++i) {
-        if (infer_request->output_tensors[i]) {
-            TF_DeleteTensor(infer_request->output_tensors[i]);
-        }
+        task->inference_done++;
+        tf_free_request(infer_request);
+        av_freep(&outputs);
+        ff_safe_queue_push_back(tf_model->request_queue, request);
+        return (task->inference_done == task->inference_todo) ? DNN_SUCCESS : DNN_ERROR;
     }
-    task->inference_done++;
-    tf_free_request(infer_request);
-    av_freep(&outputs);
-    ff_safe_queue_push_back(tf_model->request_queue, request);
-    return (task->inference_done == task->inference_todo) ? DNN_SUCCESS : DNN_ERROR;
 }
```

Comment:
basically ok for patch: lavfi/dnn_backend_tf: Separate function for filling RequestItem.

Reply:
basically ok for patch: lavfi/dnn_backend_tf: Separate function for Completion Callback